src/json.hpp.re2c

```diff
@@ -132,15 +159,156 @@ using is_unscoped_enum =
 
 namespace detail
 {
-// very useful construct against boilerplate (more boilerplate needed than in
-// C++17: http://en.cppreference.com/w/cpp/types/void_t)
-template <typename...> struct make_void
+template <typename Json> std::string type_name(const  Json& j)
```

Comment:
`Json` is just like `basic_json<Args...>` (which cannot be written, unfortunately, since it has a lot of template template parameters)

I would like to find a good name too, I'll think about it